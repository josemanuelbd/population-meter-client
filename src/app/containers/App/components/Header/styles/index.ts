import styled, { css } from 'styled-components';

import { HEADER_SIZE, HEADER_SIZE_TABLET } from 'base/styles/sizes';
import { HEADER_BORDER_COLOR, HEADER_BACKGROUND } from 'base/styles/skin';
import { TABLET } from 'base/styles/media-queries';
import getLayerPosition from 'base/styles/z-index';

export const Wrapper = styled.header``;
Wrapper.displayName = 'Wrapper';

const HeaderStyles = css`
  height: ${ HEADER_SIZE }rem;
  width: 100%;
  z-index: ${ getLayerPosition('HEADER') };

  ${ TABLET } {
    height: ${ HEADER_SIZE_TABLET }rem;
  }
`;

export const HeaderBar = styled.div`
  ${ HeaderStyles }
  align-items: center;
  background-color: ${ HEADER_BACKGROUND };
  border-bottom: 1px solid ${ HEADER_BORDER_COLOR };
  display: flex;
  padding: 1rem;
  position: fixed;
  top: 0;
`;
HeaderBar.displayName = 'HeaderBar';

export const HeaderSpacer = styled.div`
  ${ HeaderStyles }
`;
HeaderSpacer.displayName = 'HeaderSpacer';
