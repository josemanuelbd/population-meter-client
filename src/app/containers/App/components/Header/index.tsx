import React from 'react';

import { Wrapper, HeaderBar, HeaderSpacer } from './styles';

const Header: React.FC = () => (
  <Wrapper>
    <HeaderBar>Fixed Header</HeaderBar>
    <HeaderSpacer />
  </Wrapper>
);

export default Header;
