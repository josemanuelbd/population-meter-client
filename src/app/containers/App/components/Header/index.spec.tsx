import React from 'react';
import { shallow } from 'enzyme';

import Header from '.';

describe('Header', () => {
  it('should return a valid header', () => {
    const component = shallow(<Header />);
    expect(component).toMatchSnapshot();
  });
});
