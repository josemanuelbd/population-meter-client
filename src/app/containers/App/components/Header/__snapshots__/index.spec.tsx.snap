// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should return a valid header 1`] = `
<Wrapper>
  <HeaderBar>
    Fixed Header
  </HeaderBar>
  <HeaderSpacer />
</Wrapper>
`;
